src/cmds/qsub.c

```diff
@@ -3596,9 +3600,9 @@ get_script(FILE *file, char *script, char *prefix)
 
 #ifdef WIN32
 
-	_snprintf(tmp_name, MAXPATHLEN, "%s\\%s", tmpdir, tmp_template);
+	_snprintf(tmp_name, sizeof(tmp_name), "%s\\%s", tmpdir, tmp_template);
 	if ((in = _mktemp(tmp_name)) != NULL) {
-		snprintf(script, MAXPATHLEN, "%s", tmp_name);
+		snprintf(script, MAXPATHLEN + 1, "%s", tmp_name);
```

Comment:
Because "script" was passed in as a pointer. The callers of get_script() (all two of them) happen to pass a pointer to script_tmp, which is a global. This might not always be the case as they could pass a pointer to a malloc'd region. That would prevent the compiler from determining the correct size.

As an alternative, we could add another parameter to get_script() that would contain the size of the buffer. I was trying to avoid adding more code whenever possible since the PR is already quite large.